cardano-cli/src/Cardano/CLI/Commands/EraBased.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Cardano.CLI.Commands.EraBased
+  ( EraBasedCommand (..)
+  , AnyEraCommand (..)
+  , renderEraBasedCommand
+  , pAnyEraCommand
+  , pEraBasedCommand
+  ) where
+
+import           Cardano.Api (CardanoEra (..))
+
+import           Cardano.CLI.Environment
+import           Cardano.CLI.Options.Common
+import           Cardano.CLI.Options.EraBased.Governance
+
+import           Data.Foldable
+import           Data.Text (Text)
+import           Options.Applicative (Parser)
+import qualified Options.Applicative as Opt
+
+data AnyEraCommand where
+  AnyEraCommandOf :: CardanoEra era -> EraBasedCommand era -> AnyEraCommand
+
+data EraBasedCommand era
+  = EraBasedGovernanceCmd (EraBasedGovernanceCmd era)
+
+renderEraBasedCommand :: EraBasedCommand era -> Text
+renderEraBasedCommand = \case
+  EraBasedGovernanceCmd cmd -> renderEraBasedGovernanceCmd cmd
+
+pAnyEraCommand :: EnvCli -> Parser AnyEraCommand
+pAnyEraCommand envCli =
+  asum
+    [ -- Note, byron is ommitted because there is already a legacy command group for it.
+
+      subParser "shelley"
+        $ Opt.info (AnyEraCommandOf ShelleyEra <$> pEraBasedCommand envCli ShelleyEra)
+        $ Opt.progDesc "Shelley era commands"
+    , subParser "allegra"
+        $ Opt.info (AnyEraCommandOf AllegraEra <$> pEraBasedCommand envCli AllegraEra)
+        $ Opt.progDesc "Allegra era commands"
+    , subParser "mary"
+        $ Opt.info (AnyEraCommandOf MaryEra <$> pEraBasedCommand envCli MaryEra)
+        $ Opt.progDesc "Mary era commands"
+    , subParser "alonzo"
+        $ Opt.info (AnyEraCommandOf AlonzoEra <$> pEraBasedCommand envCli AlonzoEra)
+        $ Opt.progDesc "Alonzo era commands"
+    , subParser "babbage"
+        $ Opt.info (AnyEraCommandOf BabbageEra <$> pEraBasedCommand envCli BabbageEra)
+        $ Opt.progDesc "Babbage era commands"
+    , subParser "conway"
+        $ Opt.info (AnyEraCommandOf ConwayEra <$> pEraBasedCommand envCli ConwayEra)
+        $ Opt.progDesc "Conway era commands"
+
+    , subParser "latest"
+        $ Opt.info (AnyEraCommandOf BabbageEra <$> pEraBasedCommand envCli BabbageEra)
+        $ Opt.progDesc "Latest era commands (Babbage)"
+    , subParser "experimental"
```

Comment:
I'm skeptical of having an experimental set of commands, it will add more churn for not that much benefit. We should just print something to `stdout` if a command or group of commands is experimental.

Reply:
`experimental` isn't strictly necessary because users wanting to test Conway can still use `conway`.  The cost of having `experimental` isn't that high though since it is effectively equivalent to `conway`.